assets/js/googlesitekit/components-gm2/ProgressBar.js

```diff
@@ -21,17 +21,50 @@
  */
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
+import {
+	BREAKPOINT_DESKTOP,
+	BREAKPOINT_SMALL,
+	BREAKPOINT_TABLET,
+	BREAKPOINT_XLARGE,
+	useBreakpoint,
+} from '../../hooks/useBreakpoint';
 
 export default function ProgressBar( {
 	className,
 	small,
 	compress,
 	indeterminate,
 	height,
+	smallHeight,
+	tabletHeight,
+	desktopHeight,
 	progress,
 } ) {
+	const breakpoint = useBreakpoint();
+
+	let progressBarHeight = height;
+
+	if ( BREAKPOINT_SMALL === breakpoint && smallHeight ) {
+		progressBarHeight = smallHeight;
+	}
+
+	if ( BREAKPOINT_TABLET === breakpoint && tabletHeight ) {
+		progressBarHeight = tabletHeight;
+	}
+
+	if (
+		( BREAKPOINT_XLARGE === breakpoint ||
+			BREAKPOINT_DESKTOP === breakpoint ) &&
+		desktopHeight
+	) {
+		progressBarHeight = desktopHeight;
+	}
+
 	const margin =
-		typeof height !== 'undefined' ? Math.round( height / 2 ) : undefined;
+		typeof progressBarHeight !== 'undefined'
+			? // 4px is the height of the progress bar.
+			  Math.round( ( progressBarHeight - 4 ) / 2 )
```

Comment:
Hmm, if `progressBarHeight` is less than `4` we'll end up with a negative margin. Maybe we should guard against `progressBarHeight` being `< 4` with a call to `invariant()`...